inc/Menu/Item/Revert.php

```diff
@@ -7,20 +7,33 @@
  *
  * Quick revert to the currently shown page revision
  */
-class Revert extends AbstractItem {
+class Revert extends AbstractItem
+{
 
     /** @inheritdoc */
-    public function __construct() {
+    public function __construct()
+    {
         global $REV;
         global $INFO;
+        global $INPUT;
         parent::__construct();
 
-        if(!$INFO['ismanager'] || !$REV || !$INFO['writable']) {
+        // if we are comparing with the current version then
+        // assume the other version is what we want to revert to
+        $comparedWithCurrent = in_array('current', $INPUT->ref('rev2')) || in_array('', $INPUT->ref('rev2'));
```

Comment:
I wonder where rev2's `''` value come from.

Reply:
I'm not sure why, but when you first load the compare page, `$INPUT->ref('rev2')` is an array with [0] = 'current' and [1] = '$timestampOfRevision'. When you toggle between inline and side by side the order is swapped and current is replaced with ''. 

If you're viewing 2 non-current versions then the order is swapped, but they both remain as timestamps for the given revision.

Reply:
@phy25 see for example in the html.php https://codesearch.dokuwiki.org/search?project=dokuwiki&q=rev2&defs=&refs=&path=&hist=&type=

I think the simplest summary is that rev2 is appearing if you are navigating via the diff view.

Reply:
A swap makes it `''`... Interesting!